developer_setup.md

```diff
@@ -1,323 +1,213 @@
+sudo PGSETUP_INITDB_OPTIONS='--auth trust --username root --encoding UTF-8 --locale C' postgresql-setup --initdb
 ```
 
-For provider, UI or other plugin development, see [the guide on that topic](developer_setup/plugins.md).
+On Debian and Ubuntu, configure a cluster using `pg_createcluster`.
 
-#### Some troubleshooting notes
+```bash
+sudo pg_dropcluster --stop 10 main
+sudo pg_createcluster -e UTF-8 -l C 10 main -- --auth trust --username root
+```
 
-* First login fails
+On macOS, configured a cluster using `initdb` directly.
 
-Make sure you have memcached running. If not stop the server with `bundle exec rake evm:stop`,
-start memcached and retry.
+```bash
+rm -rf /usr/local/var/postgres
+initdb --auth trust --username root --encoding UTF-8 --locale C /usr/local/var/postgres
+```
 
-* `bin/setup fails` while trying to load the gem 'sprockets/es6'
+Enable and start the service.
 
-If this happens check the log for
-`ExecJS::RuntimeUnavailable: Could not find a JavaScript runtime` a few lines down.
-When this message is present, then the you need to install `node.js` and re-try
+Using systemd:
 
-* `bin/setup` fails while trying to install the 'nokogiri' gem
+```bash
+systemctl enable --now postgresql
+```
 
-If this happens, you may be missing developer tools in your OS X. Try to install them with
-`xcode-select --install` and re-try.
+Using Homebrew:
 
-* `bin/setup` fails to install the 'sys-proctable' gem, or installs the wrong version.
+```bash
+brew services start postgresql
+```
 
-If this happens it may be a Bundler issue. Try running `bundle config specific_platform true`
-and re-try.
+## Install nvm and JavaScript build utilities
 
-* Can't install any gems during bundle install
+Yarn, Gulp and Webpack are required to compile JavaScript assets. NodeJS version 10.16.x is required. If your distribution doesn't ship NodeJS 10.x, you can install [nvm](https://github.com/nvm-sh/nvm) and follow the setup steps. Then install `yarn`, `gulp-cli` and `webpack`.
 
+```bash
+nvm install 10
+npm install -g yarn gulp-cli webpack
 ```
-# install all dependencies to a local path
-bundle install --path vendor/bundle
-```
 
-* Can't install `nokogiri-1.7.2` on a Mac
+## Install Ruby and Bundler
+
+A Ruby version manager is *strongly* recommended. Use any one of the following:
+
+* [rbenv](https://github.com/rbenv/rbenv)
+* [rvm](http://rvm.io/)
+* [chruby](https://github.com/postmodern/chruby)
 
+Using the Ruby version manager, install `ruby` >= 2.4.0 and < 2.6.0 and `bundler` >= 1.16.
+
+## Clone the project
+
+```bash
+git clone git@github.com:ManageIQ/manageiq.git
+cd manageiq
 ```
-# install dependencies
-brew install pkgconfig
-brew install libxml2
 
-# test the gem can be compiled with the right libxml2
-export PKG_CONFIG_PATH=/usr/local/opt/libxml2/lib/pkgconfig
-gem install nokogiri -v 1.7.2 -- --use-system-libraries
+## Configure Project Manually
```

Comment:
I don't think we should duplicate this here

Reply:
Do you think the manual steps should be removed? I much prefer to know what is happening when I run these opaque one liners like `bin/setup`. I mean, it's clearly setting something up, but until I know what it's doing, I would rather learn all the steps and tasks myself.

Reply:
I'm mostly concerned that the steps are duplicated here and won't get updated when `bin/setup` changes.  Anyone that wants to know the details can read through the steps in `bin/setup` or `bin/update` and run them manually.

Reply:
Yes, I had that concern when writing this section too. It definitely increases the chasm between documentation and code. I'll convert that section into a subsection under the `bin/setup` section that outlines the different quirks for macOS and CentOS.